rosco-core/src/test/groovy/com/netflix/spinnaker/rosco/providers/util/ImageNameFactorySpec.groovy

```diff
@@ -142,15 +155,20 @@ class DefaultImageNameFactorySpec extends Specification {
   void "should recognize fully-qualified ubuntu package name plus extra packages"() {
     setup:
       def clockMock = Mock(Clock)
-      def imageNameFactory = new DefaultImageNameFactory(clock: clockMock)
-      def selectedOptions = new BakeOptions.Selected(baseImage: new BakeOptions.BaseImage(id: "ubuntu", packageType: "DEB"))
+      def imageNameFactory = new ImageNameFactory(clock: clockMock)
       def bakeRequest = new BakeRequest(package_name: "nflx-djangobase-enhanced_0.1-3_all kato redis-server",
                                         build_number: "12",
                                         commit_hash: "170cdbd",
                                         base_os: "ubuntu")
 
+      def osPackages = bakeRequest.package_name.split(" ").collect { packageName ->
+        PackageNameConverter.parseDebPackageName(packageName)
+      }
+
     when:
-      def (imageName, appVersionStr, packagesParameter) = imageNameFactory.deriveImageNameAndAppVersion(bakeRequest, selectedOptions)
+      def imageName = imageNameFactory.buildImageName(bakeRequest, osPackages)
+      def appVersionStr= imageNameFactory.buildAppVersionStr(bakeRequest, osPackages.head())
```

Comment:
Using `head()` in some spots and `first()` in others may result in future readers searching around to see if they behave the same. We should probably just stick to one (I prefer `first()` but don't feel that strongly about it).